### ISSUE TITLE
ID Server with Windows and Custom Claims in Database

### ISSUE THREAD

**Issue by johnfabian** [IdentityServer/IdentityServer4#1113]
I'm using web api and angular 2 in an intranet environment. I need to be able to get the current user, windows identity, then query a database to see if the user has access to the app, storing their ad account as domain\user then query a claims table which will contain roles and whatever else to add to the token. 

Should I use windows since I will be storing thier username and roles in database? I don't want them to have to login.  I also need to be able to show a custom Access Denied page. 

I need to figure out the following tasks:
-Windows Auth or some other flow without requiring login, Inside a Intranet environment but user stored in table. 
-Which flow with Angular 2 , implicit ?

**Comment by leastprivilege** [comment]
This seems to be a general question about IdentityServer - not a bug report or an issue. 

Please use StackOverflow for that. This has the advantage that questions and answers can be easily found by search engines, and that there are more people answering questions than just us.

For IdentityServer3
https://stackoverflow.com/questions/tagged/?tagnames=identityserver3&sort=newest

For IdentityServer4
https://stackoverflow.com/questions/tagged/?tagnames=identityserver4&sort=newest

For commercial support options - visit
https://identityserver.io

**Comment by johnfabian** [comment]
Ok sorry about that, I have asked there and gitter and can't s seem to get any answers other than enable windows auth on the service and call to roles on every request which is bogus so guess I'll figure it out myself or fall back to using WCF with WIF and SAMl since i have that working.  Thanks

**Comment by leastprivilege** [comment]
What's bogus about that?

Enable Windows authN and load your roles from a database when you collect the claims for the token...

**Comment by johnfabian** [comment]
That's what I was doing, I meant I kept getting answers to not use token to lookup the roles on every request which isn't what I want to do, I want to use tokens with application based roles. I actually figured it out after 3 days of trial and error and googling, it came down to not being able to access the roles on the server side with the Authorize attribute which I took away and just looked at the claims that were being passed and noticed the ClaimType was different and changed it to use the .net claimtype and it started working as expected. 

I wish I could figure out how to get it to work with just the "role" claim type instead of the long version of the .net claim type. Even when I add the "role" claim type it still changes it to be the .net ClaimType.  Name no problem it pass through with the identity server name claim type but role doesn't but it's at  least working.

Thank you for your reply. Danke!!
